specs/bep-v1.rst

```diff
@@ -62,468 +62,274 @@ another.
-    |                         Length of Key                         |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    /                                                               /
-    \                     Key (variable length)                     \
-    /                                                               /
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                        Length of Value                        |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    /                                                               /
-    \                    Value (variable length)                    \
-    /                                                               /
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-
-Fields (ClusterConfigMessage)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. Documentation note: the first time a field is mentioned it is put in **bold
-   text**. We use the Space Separated names in running text and ASCII art
-   diagrams, and CamelCase in the XDR syntax block at the end.
-
-The **Folders** field contains the list of folders that will be synchronized
-over the current connection.
+    message ClusterConfig {
+        repeated Folder folders = 1;
+    }
 
-The **Options** field is a list of options that apply to the current
-connection. The options are used in an implementation specific manner. The
-options list is conceptually a map of keys to values, although it is
-transmitted in the form of a list of key and value pairs, both of string type.
-Key ID:s are implementation specific. An implementation MUST ignore unknown
-keys. An implementation MAY impose limits on the length keys and values. The
-options list may be used to inform devices of relevant local configuration
-options such as rate limiting or make recommendations about request
-parallelism, device priorities, etc. An empty options list is valid for
-devices not having any such information to share. Devices MAY NOT make any
-assumptions about peers acting in a specific manner as a result of sent
-options.
-
-
-Fields (Folder Structure)
-~~~~~~~~~~~~~~~~~~~~~~~~~
+    message Folder {
+        string id                   = 1;
+        string label                = 2;
+        bool   read_only            = 3;
+        bool   ignore_permissions   = 4;
+        bool   ignore_delete        = 5;
+        bool   disable_temp_indexes = 6;
 
-The **ID** field contains the folder ID, as a human readable string.
+        repeated Device devices = 16;
+    }
 
-The **Label** field contains the folder label, as human readable name for the folder.
+    message Device {
+        bytes           id                = 1;
+        string          name              = 2;
+        repeated string addresses         = 3;
+        Compression     compression       = 4;
+        string          cert_name         = 5;
+        int64           max_local_version = 6;
```

Comment:
Seems like this should be uint64, but I haven't actually researched if there's some backwards compatibility issue.